Adding functions to subject model

yaelnoyman1 opened the pull request:
Updating modles.py and create test_subject.py
Resolve #34

yaelnoyman1 committed:
Adding functions to subject model

Updating modle.py and create test_subject.py

NitzanMordhai reviewed (changes requested):
fix typo in the commit (modle.py --> models.py)

Shalevro2 reviewed (approved):
LGTM

Yarboa commented:
> fix typo in the commit (modle.py --> models.py)

@yaelnoyman1 ^^

Yarboa reviewed (changes requested):
please use fixtures in tests, for creating the data for the tests,

Consult with @Anas-mhameed look at his PR

yaelnoyman1 committed:
Adding fixtures in test_subject.py

yaelnoyman1 committed:
fix test_subject.py

Yarboa commented on projboard/tests/test_subject.py:
```diff
@@ -7,36 +7,43 @@
     @pytest.mark.django_db
     def test_create_subject(db):
-        subject = Subject.create_new_subject(name=SPORT)
+        subject = Subject.create_new_subject(SPORT)
```
Consider this please 
 @pytest.mark.django_db
@pytest.fixture
def generate_subject(name='name'):
    subject = Subject(name=name)
    subject.save()
    return subject
    
    
    It is called as following:
     def test_create_subject(generate_subject):
        '''Here you have to search in database the subject'''
        
        assert generate_subject.name==<to the returned subject from db>
        Please consult others, to understand how to write the tests

yaelnoyman1 replied:
I changed it to this, is that ok?

@pytest.fixture
@pytest.mark.django_db
def generate_subject():
    subject1 = Subject.create_and_get_subject(name="yael")
    subject2 = Subject.create_and_get_subject(name="gal")
    subject1.save()
    subject2.save()
    return [subject1, subject2]

Yarboa commented on projboard/tests/test_subject.py:
```diff
@@ -7,36 +7,43 @@
-        Subject.create_new_subject(name=SPORT)
-        subject = Subject.edit_subject(SPORT, ANIMALS)
+    def test_update_subject(self, generate_subject):
+        Subject.create_new_subject(generate_subject)
```
Please review test material again
Please review all tests

yaelnoyman1 replied:
^^

Yarboa replied:
@yaelnoyman1 I assume this is done
Correct?

yaelnoyman1 replied:
Yes

avi-biton commented on projboard/models.py:
```diff
@@ -19,11 +19,68 @@ def get_subject_by_name(name):
+        return subjects_list
+
+    @staticmethod
+    def create_new_subject(name):
```
This method is not only creating a new subject, it also returns an existing subject (if exist).  
Consider renaming the method to `get_subject`

yaelnoyman1 replied:
I changed to "create_and_get_subject"

avi-biton commented on projboard/models.py:
```diff
@@ -19,11 +19,68 @@ def get_subject_by_name(name):
+            return subject
+
+    @staticmethod
+    def edit_subject(existing_name, new_name):
```
Consider renaming the method to `rename_subject` as it is not editing the subject, its just renaming it.

Yarboa replied:
Any updates to this?

yaelnoyman1 replied:
Done

Yarboa replied:
@yaelnoyman1 Please reply here

yaelnoyman1 replied:
I changed the name to "rename_suject"

yaelnoyman1 committed:
Fix models.py and test_subject.py

in models.py i changed the name of the function "create_new_subject" to "creare_and_get_subject"
and in test_subject.py i added a fixture and i changed the functions.

Shalevro2 reviewed (approved):
LGTM

Yarboa commented on projboard/tests/test_subject.py:
```diff
@@ -0,0 +1,45 @@
+        assert not subject
+
+    @pytest.mark.django_db
+    def test_get_list_subjects_names(self, generate_subject):
```
Consider this
listSubject = Subject.get_list_subjects_names()
for i in generate_subject:
     assert subject.name in listSubject

yaelnoyman1 replied:
Done

Yarboa commented on projboard/tests/test_subject.py:
```diff
@@ -0,0 +1,45 @@
+        assert not subject
+
+    def test_update_subject(self, generate_subject):
+        subject = generate_subject[0]
```
Can you describe what you test here?

yaelnoyman1 replied:
I now see that line 32 is not necessary
In line 33 I change the existing name to another name and in line 34 I check if it has changed

Yarboa replied:
Closing this

yaelnoyman1 committed:
Fix again models.py and test_subject.py

in models.py i changed the name of the function "edit_subject" to "rename_subject"
and in test_subject.py i fix the function "test_get_list_subjects_names".

yaelnoyman1 committed:
Changing  functions

"test_create_subject" and "test_delete_subject"

Shalevro2 commented:
Hi @yaelnoyman1,

I think you need to organize your code according #40, #30, #43, #41.
Take a look in the PR's above. We split models.py to directory that called models, according to this change you need also to change all the imports in the migrations files and tests.

finziamit commented:
> Hi @yaelnoyman1,
> 
> I think you need to organize your code according #40, #30, #43, #41. Take a look in the PR's above. We split models.py to directory that called models, according to this change you need also to change all the imports in the migrations files and tests.

+1
Please separate the subject from models.py file to projboard/models/subject.py (also relocate the __init__.py file to models repository) and fix the imports

Yarboa commented:
Please rebase main 
resolve conflict and push --force

yaelnoyman1 committed:
Organizing the models

Adding models directory
Inserting and creating models files- article.py, user.py, subject.py
Updating the imports in the migrations files.

yaelnoyman1 commented:
> Please rebase main resolve conflict and push --force

Done

yaelnoyman1 committed:
Fix "test_get_list_subject_names"

Yarboa commented on projboard/tests/test_subject.py:
```diff
@@ -0,0 +1,40 @@
+
+@pytest.fixture
+@pytest.mark.django_db
+def generate_subject():
```
Please use two fixtures

If you need 1 use fixture for one if you need more, use another fixture, Please review all the tests

def generate_subject():
    subject = Subject.create_and_get_subject(name="yael")
    subject.save()
    return subject

and another one  to return more
def generate_subjects(generate_subject)
  subject = Subject.create_and_get_subject(name="yael")
    subject.save()
    return [subject, generate_subject]

yaelnoyman1 replied:
Done

Yarboa replied:
@yaelnoyman1 I assume it is done
Please reply on the conversations

yaelnoyman1 replied:
I changed the fixture like you asked
and changed the functions

yaelnoyman1 committed:
fix instead  of one fixtures, to two fixtures

Anas-mhameed commented:
I suggest you squash all commits into just one commit

yaelnoyman1 commented:
> fix typo in the commit (modle.py --> models.py)

done

yaelnoyman1 commented:
> Hi @yaelnoyman1,
> 
> I think you need to organize your code according #40, #30, #43, #41. Take a look in the PR's above. We split models.py to directory that called models, according to this change you need also to change all the imports in the migrations files and tests.

Done, And I did rebase main
